ansible_rulebook/engine.py

```diff
@@ -350,6 +344,17 @@ async def run_ruleset(
             continue
         results = []
         try:
+            # Asset added by other ruleset, not related to current event
```

Comment:
@bzwei Each ruleset is running in independent sessions so which other ruleset does this relate to? The rulesets communicate with each other by doing a post_event/set_fact with the name of the ruleset. For a running ruleset it doesn't know if the data was generated by a source plugin or another running ruleset from the rulebook.

Reply:
When ruleset1 sets a fact to ruleset2, it may add a new action to the current plan of the ruleset2 after ruleset2 already finished processing all items in the plan. This code gives it a chance to repeat the processing of the plan before it starts a new plan.